core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -312,6 +312,8 @@ object KafkaConfig {
   val ShareGroupHeartbeatIntervalMsProp = "group.share.heartbeat.interval.ms"
   val ShareGroupMinHeartbeatIntervalMsProp = "group.share.min.heartbeat.interval.ms"
   val ShareGroupMaxHeartbeatIntervalMsProp = "group.share.max.heartbeat.interval.ms"
+  val ShareGroupRecordLockDurationMsProp = "group.share.record.lock.duration.ms"
+  val ShareGroupMaxRecordLockDurationMsProp = "group.share.record.lock.duration.max.ms"
```

Comment:
@AndrewJSchofield These seems confusing to me, just re-read the KIP. Can't we have just one config which is `group.share.record.lock.duration.ms` and has minimum and maximum, why do we require 2 of them?

Reply:
One is the duration of the record locks for a group which does not have its own configuration for lock duration, and the other is the maximum value that is permitted for the group configuration for lock duration.

Reply:
Though it must be something you already thought about and I am sure I am missing something. But what about a single configuration which has a min, max and a default value.

Is it more about that `group.share.record.lock.duration.ms` configuration cannot be more that the `group.share.record.lock.max.duration.ms`, I think that makes sense. In that case we do not need to define `group.share.record.lock.duration.ms` in KafkaConfig as that should be part of consumer config, correct?

Reply:
Yeah, I think it's latter and KIP says that `group.share.record.lock.duration.ms` config is specified by consumer clients.

Reply:
I am not sure, is there any change expected in the code?